### PR TITLE
[Dam] Update interfaces of Element, Condition and Scheme

### PR DESCRIPTION
**Pull request by joaquinirazabal**
**Description**
Mainly, what is requested in issue #7381 for the DamApplication

**Changelog**
- Adding const to the ProcessInfo that is passed to the functions.
- GetValuesVector, GetFirstDerivativesVector, GetSecondDerivativesVector and Check functions in Elements and Conditions are const functions now.
-  GetValueOnIntegrationPoints is deprecated in favor of CalculateOnIntegrationPoints.

### CODE REVIEW

File: applications/DamApplication/custom_conditions/UP_condition.cpp [hunk 1]
```diff
@@ -26,22 +26,21 @@ Condition::Pointer UPCondition<TDim,TNumNodes>::Create(IndexType NewId,NodesArra
 //----------------------------------------------------------------------------------------
 
 template< >
-void UPCondition<2,2>::GetDofList(DofsVectorType& rConditionDofList, ProcessInfo& rCurrentProcessInfo)
+void UPCondition<2,2>::GetDofList(DofsVectorType& rConditionDofList, const ProcessInfo& rCurrentProcessInfo) const
 {
     KRATOS_TRY
-    
-    GeometryType& rGeom = GetGeometry();
```
Comment (philbucher) [inline applications/DamApplication/custom_conditions/UP_condition.cpp:33]: if you make this `const` you wouldn't need the other changes

File: applications/DamApplication/custom_strategies/schemes/dam_P_scheme.hpp [hunk 2]
```diff
@@ -366,7 +366,7 @@ class DamPScheme : public Scheme<TSparseSpace,TDenseSpace>
         Condition::Pointer rCurrentCondition,
```
Comment (philbucher) [inline applications/DamApplication/custom_strategies/schemes/dam_P_scheme.hpp:366]:
```suggestion
        Condition& rCurrentCondition,
```

the scheme functions also changed, but you can make that change also in a followup PR

**Review by philbucher — APPROVED** [review]
:+1:
looks good to me, but the `Scheme` functions need some more attention. In this PR or the next :)

**Comment by joaquinirazabal** [comment]
Thanks, @philbucher for your comments, I will make the changes in another PR.

**Pull request merged**